test/rules.jl

```diff
@@ -132,11 +134,11 @@ end
         fx, f_pushforward = res
         df(Δx, Δp) = f_pushforward(NamedTuple(), Δx, Δp)
 
-        df_dx::Thunk = df(One(), Zero())
-        df_dp::Thunk = df(Zero(), One())
+        df_dx = df(One(), Zero())
+        df_dp = df(Zero(), One())
```

Comment:
why don't we want to return `Thunk`s here?

Reply:
This was really an edge case before, because only single-arg `+` doesn't unthunk `Thunk`s, so only the one-arg functions returned `Thunk`s. For `chain`, we need to unthunk `Thunk`, since they could still contain `AbstractWirtinger`, so I don't think we want to return `Thunk` here.